### PR TITLE
[2.0] Remove simple references leftovers

### PR DESCRIPTION
**Pull request by malarzm**
We need to decide whether we still want to call references using `storeAs="id"` "simple" or not (I would be in favour of doing so, feels more natural than "references stored as ids" however I may just be used to "simple references" term)

### CODE REVIEW

**Comment by alcaeus** [comment]
I've started calling them "ID references", opposed to "DBRef references". Since there's no more mention to "simple" in the code, I'd like to avoid that term in the future since it might make it more complex to understand for newbies.

**Comment by coudenysj** [comment]
I like the @alcaeus approach :)

**Comment by malarzm** [comment]
Changed "simple references" occurrences to "ID references" as suggested :+!:

File: tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php [hunk 1]
```diff
@@ -1,68 +1,68 @@
...
+        $a->addRef($b);
+        $a->addRef($c);
+
+        $this->dm->persist($a);
+        $this->dm->persist($b);
+        $this->dm->persist($c);
+        $this->dm->flush();
+        $this->dm->clear();
+
+        $collection = $this->dm->getDocumentCollection(__NAMESPACE__ . '\GH499Document');
```
Comment (alcaeus) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php]: You can use the `GH499Document::class` here.

File: tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php [hunk 2]
```diff
@@ -1,68 +1,68 @@
...
+
+        $a = $collection->findOne(array('_id' => new \MongoId($a->getId())));
```
Comment (alcaeus) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php]: Short array syntax

File: tests/Doctrine/ODM/MongoDB/Tests/Mapping/Driver/fixtures/yaml/TestDocuments.User.dcm.yml [hunk 3]
```diff
@@ -1,44 +1,44 @@
...
-    groups:
-      targetDocument: Documents\Group
-      cascade: all
-  lifecycleCallbacks:
-    prePersist: [doStuffOnPrePersist]
-    postPersist: [doStuffOnPostPersist, doOtherStuffOnPostPersist]
-  alsoLoadMethods:
-    doStuffOnAlsoLoad: unmappedField
-
+TestDocuments\User:
```
Comment (alcaeus) [inline tests/Doctrine/ODM/MongoDB/Tests/Mapping/Driver/fixtures/yaml/TestDocuments.User.dcm.yml]: I see Windows line endings here.

File: tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php (hunk 1 above)
Comment (malarzm) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php]: I'm more wondering why is this marked as an addition, it should only be mapping changed...

Comment (alcaeus) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php]: Looking at the [diff without whitespaces](https://github.com/doctrine/mongodb-odm/pull/1480/files?w=0) hides the lines, so I assume there were changes to the line endings.

Comment (malarzm) [inline tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php]: Funny, the only thing that happened to this files was a rebase, will revisit my git settings and fix it :+1:

**Comment by malarzm** [comment]
@alcaeus I've fixed CRLF mess that I introduced

> I think the changes to the mapping in tests (changing simple to storeAs) could be backported to the master branch, what do you think?

I've split commits into fixing test mappings and removing leftovers for cherry picking 👍

**Pull request merged**